### PR TITLE
232 magiccdata hard coding

### LINKED ISSUE #232: MAGICCData append is hard-coded
**Describe the bug**

MAGICCData append is hard-coded to be in place.

**Failing Test**

```python
original = MAGICCData("RCP26.SCEN")
copy = MAGICCData("RCP26.SCEN")
other_scen = MAGICCData("RCP45.SCEN")
original.append(other_scen, inplace=False)
assert original == copy
```

**Expected behavior**

`inplace` should be passed onto other `append` calls. 

**Screenshots**

**System (please complete the following information):**

 - OS: Linux
 - Python and Pymagicc version: Python 3.7

**Additional context**

### PR DESCRIPTION
**Pull request by znicholls**
# Pull request

Please confirm that this pull request has done the following:

- [x] Tests added
- [x] Documentation added (where applicable)
- [x] Example added (either to an existing notebook or as a new notebook, where applicable)
- [x] Description in ``CHANGELOG.rst`` added

## Adding to CHANGELOG.rst

Please add a single line in the changelog notes similar to one of the following:

```
- (`#XX <http://link-to-pr.com>`_) Added feature which does something
- (`#XX <http://link-to-pr.com>`_) Fixed bug identified in (`#XX <http://link-to-issue.com>`_)
```

Closes #232

### CODE REVIEW

File: tests/test_io.py [hunk 1]
```diff
@@ -2704,21 +2705,27 @@ def test_magicc_data_append(mock_read_and_return_metadata_df):
...
     expected.columns = pd.Index(
         [datetime.datetime(tindex_yr, 1, 1, 0, 0, 0)], dtype="object"
     )
 
     pd.testing.assert_frame_equal(
-        mdata.timeseries(),
+        res.timeseries(),
         expected.sort_index().reorder_levels(mdata.timeseries().index.names),
         check_like=True,
     )
```
Comment (lewisjared) [inline tests/test_io.py]: Add a check to check that the original mdata hasn't changed for inplace=False

Comment (znicholls) [reply tests/test_io.py]: how do we do comparisons for `MAGICCData` objects?

Comment (lewisjared) [reply tests/test_io.py]: Perhaps keep a copy of the original timeseries, do the append and check that they are the same
```
orig_timeseries = mdata.timeseries().copy()

...

if not inplace:
      pd.testing.assert_frame_equal(
        orig_timeseries,
        mdata.timeseries(),
        check_like=True,
    )
```

Perhaps ScmDataFrame needs an __eq__ operator or a helper?

Comment (znicholls) [reply tests/test_io.py]: > Perhaps ScmDataFrame needs an **eq** operator or a helper?

Having just read about these I am nervous, equality is a tricky thing to define. I like leaving it up to the users to decide what is equal enough (just `timeseries`, `timeseries` and `metadata` etc.)

**Pull request merged**